interfaces/cython/cantera/onedim.py

```diff
@@ -169,6 +169,36 @@ def set_profile(self, component, positions, values):
         """
         super().set_profile(self.flame, component, positions, values)
 
+    def get_species_reaction_sensitivities(self, species: str, ix: int):
+        r"""
+        Compute the normalized sensitivities of a species
+        :math:`S_u` with respect to the reaction rate constants :math:`k_i`:
+
+        .. math::
+            s_i = \frac{k_i}{X} \frac{dX}{dk_i}
+        :param species:
+            Species of interest for sensitivity calculation (i.e "NO")
+        :param ix:
+            grid point index (0 is the first, -1 is the last point in the domain)
+        """
+
+        def g(sim):
+            return self.X[self.gas.species_index(species), ix]
+
+        Nvars = sum(D.n_components * D.n_points for D in self.domains)
```

Comment:
```suggestion
        n_vars = sum(dd.n_components * dd.n_points for dd in self.domains)
```
Uppercase letters should be avoided (Cantera mostly follows the PEP8 Python convention with the exception of thermodynamic properties, i.e. `T`, `P`, `X`/`Y` etc.). There are a couple of other instances above and below. Single-letter variable/function names should likewise be avoided.

Reply:
@ischoegl thank you for the feedback. 
- I corrected the formatting. 
- I'm building a test_sensitivities2 function in the pytest suite. It seems that different 1D flames have different domain/n_vars definitions, so I'm just working on the function to make sure that all different flame types are correctly accounted for. 

I apologise it's taking a bit longer than anticipated, but I'd like to make sure we have a solution that works for all flame types.

Reply:
The indexing in the solution vector varies across classes (burner, inlet, fuel_inlet), which are the various boundary domains. However, my current understanding is that the n_components term for all inlet boundaries is always zero. Therefore, rather than write multiple versions of the function, I have removed this term from i_X. 

I am having some issues writing a pytest. Comparing the fwd and adjoint method was the approach for the flame speed sensitivity function, however, there are some differences between the fwd and adjoint approaches when testing multiple conditions. Its just a rough run, but I've put the plots side by side. 

[compare_fwd_adjoint.pptx](https://github.com/user-attachments/files/18389910/compare_fwd_adjoint.pptx)

Reply:
Ok all done @ischoegl ! Ready for review!

- [ ] example file: added as samples/species_sensitivity
- [ ] function formatting:  corrected for n_vars, dd
- [ ] test_onedim.py: added adjoint sensitivity test 2

Please note that the new testing criteria are that 7 of the top 10 reactions should match between the two methods for the tested condition. There are some differences in magnitude and position between the forward and adjoint methods, and also to the CHEMKIN result as given in the example paper. However, why and how this is different is beyond my current range of capabilities as a "first-issue contribution". The new suggested test criteria is therefore as follows: 

        # Assert that at least 8 reactions match
        assert len(common_reactions) >= 8, f"Only {len(common_reactions)} Fwd and adjoint based species sensitivities do not match"
        
Any questions, issues, please let me know.